api/api.go

```diff
@@ -57,9 +67,13 @@ func New(db *database.DB, mf *metafetcher.MetaFetcher, logger *logrus.Logger, ma
 			Storage:           t.Storage,
 		}
 	}
+	if deps == nil {
+		deps = &lib.ProductionDependencies{}
```

Comment:
We usually have 2 constructors to avoid these conditional initialisations. 
`NewCustom` which accepts a dependency and `New` which calls `NewCustom` with the production dependency.
In testing we call `NewCustom` to inject dependencies and in production we just call `New`. That way you don't end up with a ton of calls that have `nil` passed in.